source/common/network/utility.cc

```diff
@@ -292,14 +296,35 @@ Address::InstanceConstSharedPtr Utility::getOriginalDst(int fd) {
 #ifdef SOL_IP
   sockaddr_storage orig_addr;
   socklen_t addr_len = sizeof(sockaddr_storage);
-  int status = getsockopt(fd, SOL_IP, SO_ORIGINAL_DST, &orig_addr, &addr_len);
 
-  if (status == 0) {
-    // TODO(mattklein123): IPv6 support. See github issue #1094.
-    ASSERT(orig_addr.ss_family == AF_INET);
+  int socket_domain;
+  socklen_t domain_len = sizeof(socket_domain);
+  int status = getsockopt(fd, SOL_SOCKET, SO_DOMAIN, &socket_domain, &domain_len);
+  if (status != 0) {
+    return nullptr;
+  }
+
+  switch (socket_domain) {
+  case AF_INET:
+    status = getsockopt(fd, SOL_IP, SO_ORIGINAL_DST, &orig_addr, &addr_len);
```

Comment:
Do you want to use https://github.com/envoyproxy/envoy/blob/master/source/common/api/os_sys_calls_impl.h#L26? That way, we can mock `getsockopt` in tests and be able to force coverage of failure cases.

Reply:
Ack